experimental/segmenter/src/math_helper.rs

```diff
@@ -210,6 +196,14 @@ impl<'a> MatrixBorrowed<'a, 1> {
         debug_assert_eq!(self.dims, other.dims);
         unrolled_dot_1(self.data, other.data)
     }
+
+    pub fn view(&'a self, r: Range<usize>) -> MatrixBorrowed<'a, 1> {
+        MatrixBorrowed {
```

Comment:
nit: either `debug_assert` or return `Option`

Reply:
Or ideally put the dimension back in the matrix so that you can use submatrix instead of view.